documentation/puppet_server_metrics.markdown

```diff
@@ -0,0 +1,440 @@
+
+1.  Set the `enabled` parameter to true in `metrics:registries:puppetserver:reporters:graphite`:
+
+    ```
+     metrics: {
+        server-id: localhost
+        registries: {
+            puppetserver: {
+                ...
+                reporters: {
+                    ...
+                    # enable or disable Graphite metrics reporter
+                    graphite: {
+                        enabled: true
+                    }
+                }
+            }
+        }
+    }
+    ```
+
+2.  Configure the Graphite host settings in `metrics:reporters:graphite`:
+    -   **host:** The Graphite host's IP address as a string.
+    -   **port:** The Graphite host's port number.
+    -   **update-interval-seconds:** How frequently Puppet Server should send metrics ot Graphite.
+
+3.  Verify that `metrics:registries:puppetserver:reporters:jmx:enabled` is not set to false. Its default setting is true.
+
+> **Tip:** In the Grafana UI, choose an appropriate time window from the drop-down menu.
+
+### Using the sample Grafana dashboard
+
+The [sample Grafana dashboard][] provides what we think is an interesting starting point. You can click on the title of any graph, and then click **edit** to tweak the graphs as you see fit.
+
+-   **Active requests:** This graph serves as a "health check" for the Puppet Server. It shows a flat line that represents the number of CPUs you have in your system, a metric that indicates the total number of HTTP requests actively being processed by the server at any moment in time, and a rolling average of the number of active requests. If the number of requests being processed exceeds the number of CPUs for any significant length of time, your server might be receiving more requests than it can efficiently process.
+
+-   **Request durations:** This graph breaks down the average response times for different types of requests made by Puppet agents. This indicates how expensive catalog and report requests are compared to the other types of requests. It also provides a way to see changes in catalog compilation times when you modify your Puppet code. A sharp curve upward for all of the types of requests indicates an overloaded server, and they should trend downward after reducing the load on the server.
+
+-   **Request ratios:** This graph shows how many requests of each type that Puppet Server has handled. Under normal circumstances, you should see about the same number of catalog, node, or report requests, because these all happen once per agent run. The number of file and file metadata requests correlate to how many remote file resources are in the agents' catalogs.
+
+-   **Communications with PuppetDB:** This graph tracks the amount of time it takes Puppet Server to send data and requests for common operations to, and receive responses from, PuppetDB.
+
+-   **File Sync:** This graph tracks how long Puppet Server spends on File Sync operations, for both its storage and client services.
```

Comment:
Might be best to leave this out or tag it as "for PE only" since these ones aren't available in OSS.